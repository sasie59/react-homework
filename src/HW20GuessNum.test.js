import React from 'react';
import { getByTestId, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW20GuessNum';

global.Math.random = () => 0.240;

test('should HW20GuessNum', () => {
  const { container } = render(<HW />);
  const input = container.querySelector('input');
  userEvent.type(input, '25{enter}');
  expect(container).toMatchSnapshot();
});
test('should HW20GuessNum', () => {
  const { container } = render(<HW />);
  const button = container.querySelector('button');
  const input = container.querySelector('input');
  userEvent.type(input, '25');
  userEvent.click(button);
  expect(container).toMatchSnapshot();
});

test('should reset game', () => {
  const { container,getByTestId } = render(<HW />);
  const input = container.querySelector('input');
  userEvent.type(input, '241{enter}');
  expect(getByTestId('answer').textContent).toBe('答對了!答案就是241重新一局');

  userEvent.click(getByTestId('reset'));
  expect(getByTestId('range').textContent).toBe('現在範圍 :1 ~ 1000');
});

