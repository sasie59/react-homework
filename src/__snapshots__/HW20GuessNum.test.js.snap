// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW20GuessNum 1`] = `
<div>
  <div>
    <h1
      data-testid="range"
    >
      現在範圍 :
      25
       ~ 
      1000
    </h1>
    <form>
      <input
        type="text"
      />
      <button>
        submit
      </button>
    </form>
    <div
      class="hide"
      data-testid="answer"
    >
      答對了!答案就是
      241
      <button
        data-testid="reset"
      >
        重新一局
      </button>
    </div>
  </div>
</div>
`;

exports[`should HW20GuessNum 2`] = `
<div>
  <div>
    <h1
      data-testid="range"
    >
      現在範圍 :
      25
       ~ 
      1000
    </h1>
    <form>
      <input
        type="text"
      />
      <button>
        submit
      </button>
    </form>
    <div
      class="hide"
      data-testid="answer"
    >
      答對了!答案就是
      241
      <button
        data-testid="reset"
      >
        重新一局
      </button>
    </div>
  </div>
</div>
`;
